pkg/providers/snow/apibuilder.go

```diff
@@ -57,16 +61,22 @@ func KubeadmControlPlane(clusterSpec *cluster.Spec, snowMachineTemplate *snowv1.
 		fmt.Sprintf("/etc/eks/bootstrap-after.sh %s %s", clusterSpec.VersionsBundle.Snow.KubeVip.VersionedImage(), clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host),
 	)
 
-	if err := clusterapi.SetRegistryMirrorInKubeadmControlPlane(kcp, clusterSpec.Cluster.Spec.RegistryMirrorConfiguration); err != nil {
+	osFamily := clusterSpec.SnowMachineConfig(clusterSpec.Cluster.Spec.ControlPlaneConfiguration.MachineGroupRef.Name).OSFamily()
+
+	if err := clusterapi.SetRegistryMirrorInKubeadmControlPlane(kcp, clusterSpec.Cluster.Spec.RegistryMirrorConfiguration, osFamily); err != nil {
 		return nil, err
 	}
 
-	if err := clusterapi.SetProxyConfigInKubeadmControlPlane(kcp, clusterSpec.Cluster.Spec); err != nil {
+	if err := clusterapi.SetProxyConfigInKubeadmControlPlane(kcp, clusterSpec, osFamily); err != nil {
 		return nil, err
 	}
 
-	clusterapi.CreateContainerdConfigFileInKubeadmControlPlane(kcp, clusterSpec.Cluster.Spec)
-	clusterapi.RestartContainerdInKubeadmControlPlane(kcp, clusterSpec.Cluster.Spec)
+	if osFamily == v1alpha1.Bottlerocket {
```

Comment:
ahh we already have this check here
This is where I would put all the other methods I was suggesting:
`SetRegistryMirrorInKubeadmControlPlaneForBottlerocket`...